src/v/test_utils/async.h

```diff
@@ -50,4 +51,32 @@ requires ss::ApplyReturns<Predicate, bool> ||
           });
       }));
 }
+
+// When a test expects that any background fibers should complete promptly,
+// and wants to send a barrier through all the inter-CPU queues to ensure
+// that earlier-submitted tasks have reached their destination cores already.
+//
+// This is useful in tests that know they have put the system into a state where
+// it will get to a known state once all the non-i/o-blocking tasks in flight
+// have completed, such as background release of quota units.
```

Comment:
I think it's necessary to note the other big caveat: it's not just "blocking IO tasks" what won't be waited for by this method, but also any tasks which runs long enough to suspect due to exhausting its task quota. In debug mode, I think _every_ suspend point result in actual suspension so I guess this mostly does nothing in debug mode (makes me wonder if we should have the sleep _plus_ this method, as a kind of defense in depth).

Reply:
Good point, extended the comment.